### PR TITLE
Add audit rules to k8s hardening

### PR DESCRIPTION
**Pull request by dimityrmirchev**
**What this PR does / why we need it**:
This PR adds the missing audit rules to the K8s hardening document.

cc @n-boshnakov @JordanJordanov 
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

**Review by n-boshnakov — APPROVED** [review]
/lgtm

File: website/documentation/guides/security-and-compliance/kubernetes-hardening/_index.md [hunk 1]
```diff
@@ -31,6 +31,13 @@ The Gardener team takes security seriously, which is why we mandate the Security
 |242394|Kubernetes Worker Nodes must not have the sshd service enabled| :x: | Enabled to allow debugging of network issues, but it is possible to deactivate via the [sshAccess](https://gardener.cloud/docs/gardener/api-reference/core/#core.gardener.cloud/v1beta1.SSHAccess) setting|
 |242434|Kubernetes Kubelet must enable kernel protection|:white_check_mark:|Enabled for Kubernetes v1.26 or later unless disabled explicitly via [protectKernalDefaults](https://gardener.cloud/docs/gardener/api-reference/core/#kubeletconfig)|
 |245541|Kubernetes Kubelet must not disable timeouts|:white_check_mark:| Enabled for Kubernetes v1.26 or later unless disabled explicitly via [streamingConnectionIdleTimeout](https://gardener.cloud/docs/gardener/api-reference/core/#kubeletconfig)|
+|242402|The Kubernetes API Server must have an audit log path set| :x: | It is the user's responsibility to configure an audit extension that meets the requirements of their organization. Depending on the audit extension implementation the audit logs do not always need to be written on the filesystem, i.e. when `--audit-webhook-config-file` is set and logs are sent to an audit backend.|
```
Comment (JordanJordanov) [inline website/documentation/guides/security-and-compliance/kubernetes-hardening/_index.md:34]: Shall we introduce a dedicated section `Audit Log Configurations` or we better keep the rules in `Control Plane Configuration`?

Comment (dimityrmirchev) [reply website/documentation/guides/security-and-compliance/kubernetes-hardening/_index.md:34]: I also thought about this, but I do not have strong opinion on the matter. Do you prefer for them to be separate?

Comment (JordanJordanov) [reply website/documentation/guides/security-and-compliance/kubernetes-hardening/_index.md:34]: I have a slight preference to keep them separated.

**Review by n-boshnakov — APPROVED** [review]
/lgtm

**Review by JordanJordanov — APPROVED** [review]
/lgtm

**Pull request merged**